### PR TITLE
Meeting changes

### PR DESCRIPTION
**Pull request by aleokdev**

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -3,6 +3,15 @@ use serde::{Deserialize, Serialize};
 #[cfg(feature = "schemars")]
 use schemars::JsonSchema;
 
+// for the company: schema + documentation
+// TODO better docs
+// TODO fix required *
```
Comment (aleokdev) [inline src/lib.rs]: This shouldn't be here as it has nothing to do with the format.

File: src/lib.rs [hunk 2]
```diff
@@ -21,35 +30,31 @@ pub struct Document {
...
+    /// A unique ID used to associate this detection with a specific UAS.
     /// The number itself is not relevant, it just needs to be unique per UAS.
     ///
     /// If null, means the system was not able to identify the records present along with this
     /// object with any specific UAS.
     pub uas_id: Option<u64>,
     /// A list of records associated with this detection.
     pub records: Vec<Record>,
-    /// Free-form text describing the detection set. Can be, for instance, the name present on the HMI.
+    /// (Optional) Free-form text describing the detection set. Can be, for instance, the name present on the HMI.
```
Comment (aleokdev) [inline src/lib.rs]: It's obvious that it is optional from the type.

File: src/lib.rs [hunk 3]
```diff
@@ -21,35 +30,31 @@ pub struct Document {
 #[derive(Serialize, Deserialize, Clone, Debug, PartialEq)]
 #[cfg_attr(feature = "schemars", derive(JsonSchema))]
 pub struct Detection {
-    /// An unique ID used to associate this detection with a specific UAS.
+    /// (Optional)
```
Comment (aleokdev) [inline src/lib.rs]: It's obvious that it is optional from the type.

File: src/lib.rs [hunk 4]
```diff
@@ -59,23 +64,31 @@ pub struct Record {
     pub record_number: u64,
     /// Classification of the record.
     pub classification: Classification,
-    /// Whether the alarm function of the CUAS system is active or not.
+    /// (Optional)
     /// An Alarm is defined as the function of a CUAS system alerting an Operator via the HMI and
     /// the generation of associated data in the UAS Activity Log, as a result of Declared UAS
     /// activity.
```
Comment (aleokdev) [inline src/lib.rs]: Now that we have an Alarm type, this describes the type better than the member.

File: src/lib.rs [hunk 5]
```diff
@@ -59,23 +64,31 @@ pub struct Record {
     pub record_number: u64,
     /// Classification of the record.
     pub classification: Classification,
-    /// Whether the alarm function of the CUAS system is active or not.
+    /// (Optional)
```
Comment (aleokdev) [inline src/lib.rs]: Remove this.

File: src/lib.rs [hunk 6]
```diff
@@ -59,23 +64,31 @@ pub struct Record {
...
-    pub alarm: bool,
-    /// How certainly should the alarm be on, as a value from 0 (Least likely) to 1 (Most likely).
-    #[cfg_attr(feature = "schemars", validate(range(min = 0., max = 1.)))]
-    pub alarm_certainty: f64,
+    pub alarm: Option<Alarm>,
     /// The UAS location, which may be given in one of several declaration types.
+    /// Location types are composed of a tag 't' and a contents element 'c'.
     pub location: Location,
-    /// Free form text, possibly describing the model or configuration of the UAS identified.
+    /// (Optional) Free form text, possibly describing the model or configuration of the UAS identified.
```
Comment (aleokdev) [inline src/lib.rs]: Remove the optional text.

File: src/lib.rs [hunk 7]
```diff
@@ -59,23 +64,31 @@ pub struct Record {
...
     pub identification: Option<String>,
-    /// The 3D GPS location of the CUAS recorded on this instant. Overrides the document's
+    /// (Optional) The 3D GPS location of the CUAS recorded on this instant. Overrides the document's
```
Comment (aleokdev) [inline src/lib.rs]: Remove the optional text.

File: src/lib.rs [hunk 8]
```diff
@@ -59,23 +64,31 @@ pub struct Record {
...
     /// static_cuas_location.
     pub cuas_location: Option<Position3d>,
 }
 
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, Copy)]
+#[cfg_attr(feature = "schemars", derive(JsonSchema))]
+pub struct Alarm {
```
Comment (aleokdev) [inline src/lib.rs:76]: Add docs for this type.

File: src/lib.rs [hunk 9]
```diff
@@ -92,6 +105,8 @@ pub enum Classification {
 #[serde(tag = "t", content = "c")]
 /// Location of an UAS, which may be relative to the CUAS.
 pub enum Location {
+    // All this variant descriptions are overwritten by the doc comments of their respective types
+
```
Comment (aleokdev) [inline src/lib.rs]: What does this mean?

Comment (marcasmar1) [reply src/lib.rs]: This is more related to the schema visualizer and what description it shows when the mouse hovers over an element. See [this issue](https://github.com/COURAGEOUS-isf/format-website/issues/3).

Comment (aleokdev) [reply src/lib.rs]: Remove it then. That's a visualizer issue. This should be completely independent of the visualizer.

File: src/lib.rs (hunk 4 above)
Comment (marcasmar1) [reply src/lib.rs]: That is true, but I don't understand what differences there should be between member descriptions and type descriptions in situations like this.

**Pull request merged**